File: server/models/transaction.js
const mongoose = require('mongoose');

const TransactionSchema = new mongoose.Schema({
  razorpayOrderId: String,
  razorpayPaymentId: String,
  phone: String,
  amount: Number,
  status: {type:String, enum:['created','paid','failed'], default:'created'},
  coupon: {type: mongoose.Schema.Types.ObjectId, ref:'Coupon'},
  meta: {},
  note: String
},{timestamps:true});

module.exports = mongoose.model('Transaction', TransactionSchema);

File: client/lib/models/transaction.dart
class Transaction {
  final String id;
  final String razorpayOrderId;
  final String plan;
  final int amount;
  final String status;
  final String? razorpayPaymentId;
  final String? razorpaySignature;
  final String? coupon;

  Transaction({required this.id, required this.razorpayOrderId, required this.plan, required this.amount, required this.status, this.razorpayPaymentId, this.razorpaySignature, this.coupon});

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['_id'],
      razorpayOrderId: json['razorpay_order_id'],
      plan: json['plan'],
      amount: json['amount'],
      status: json['status'],
      razorpayPaymentId: json['razorpay_payment_id'],
      razorpaySignature: json['razorpay_signature'],
      coupon: json['coupon'],
    );
  }
}
